docs/manual.md

```diff
@@ -852,9 +852,20 @@ package:
             contain-element: "4.1.0"
 ```
 
+Custom version matcher are available `version-(eq|gt|ge|lt|le)`: 
+
+```yaml
+package:
+  kernel:
+    installed: true
+    versions:
+      version-gt: "4.0.0"
```

Comment:
I followed your lead with the `BeNumerically` matcher based on the comment: 
> Golang json escapes '>', '<' symbols, so we use 'gt', 'le' instead

But if you favor this approach, I can take a look at the feasibility of having `>` and the other symbols.

Reply:
I did a few quick tests since I had some time tonight and the constraints for `go-version` and `semver` are more rigid than the simple functions (`GreaterThan`...). 

I also confirmed that `go-version` with the `NewVersion` constructor is less rigid than `semver` with the `Parse` constructor.

While this approach still doesn't fit all the versions especially since some of them completely disregard standards. I would say that I eyeballed the correct implementation without fully testing the alternatives. 

Of course, I am open to other ideas.

Reply:
I've been traveling myself so only eyeballing tbis PR on my phone so far.

Are you saying the following isn't feasible?  
```yaml
semver_constraint: ">4.0.0, <5.0.0, !=4.4.0" 
```

As for package manager support, I think they would have to be handled individualy. Ex:
```yaml
semver_constraint: ">4.0.0, <5.0.0, !=4.4.0"
rpm_constraint: "..."
deb_constraint: "..."
```

I'm almost 100% sure (from memory) that semver clashes with rpm versioning schema.

I see semver as being useful outside the scope of package versioning. Http/command and other checks might be able to leverage it in the future.. once I normalize the non-gomega matchers.

Wow that JSON comment is 4yrs ago, back when Goss was only json based. Seems go offers a toggle for "<>" now:
https://stackoverflow.com/a/39079218

Reply:
I am sorry for the delay, I didn't have any free time yesterday. Reading all of your comments from the past 2 days I understand how we weren't synced on what the functionality we are adding here.

I am fully on board with calling this Matcher a `semver_constraint` and not try to force it into working with the different package managers' version specifications. 

I will modify and clean the PR up. Thank you for your feedback!

Reply:
I have pivoted to implement the `semver-constraint` matcher.